### PR TITLE
B 16906 tac parser

### PR DESCRIPTION
**Pull request by cameroncaci**
## Summary

Added new tac package to the backend. This allows for the parsing of received transportation accounting codes exported via a .txt file.

See additional documentation here:
https://github.com/CACI-MIlMOVE/mymove-docs/blob/main/docs/backend/tget/parsing/transportation-account.md

### Struct Changes
- Added new struct for desired information from the .txt file
- Added new struct representing the .txt file

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

N/A - Awaiting upstream dependencies

### CODE REVIEW

**Review by danieljordan-caci — DISMISSED** [review]
Looks good, comments are thorough and explain through the steps well. Two thumbs up from me!

**Review by TevinAdams — DISMISSED** [review]

**Review by j-Rosenthal — DISMISSED** [review]

File: pkg/parser/tac/parse.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	for _, code := range codes {
+		if !code.ExpirationDate.Before(time.Now()) {
+			pruned = append(pruned, code)
+		}
+	}
+
+	return pruned
+}
+
+// Consolidates TACs with the same TAC value. Duplicate "Transaction", aka description, calues are combined with a delimeter of ". Additional description found: "
```
Comment (deandreJones) [inline pkg/parser/tac/parse.go]: Comment on line 93 "calues". should that say "values"?

File: pkg/parser/tac/parse.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+}
+
+// This function checks two TAC codes: one existing and one new to decide which one to keep based on their ExpirationDates
+// If the ExpirationDate is the same, it appends the transactions and maintains the first code found
+// If the ExpirationDate is different, it appends the transactions and maintains the ExpirationDate further in the future
+func overwriteDuplicateCode(existingCode models.TransportationAccountingCodeDesiredFromTRDM, newCode models.TransportationAccountingCodeDesiredFromTRDM) models.TransportationAccountingCodeDesiredFromTRDM {
+
+	// If the new code expires later, append its transaction to the existing one (if not empty), and keep the new code
+	if newCode.ExpirationDate.After(existingCode.ExpirationDate) {
+		if existingCode.Transaction != "" {
```
Comment (deandreJones) [inline pkg/parser/tac/parse.go]: for simplicity, could these if statements be combined into 1

File: pkg/parser/tac/parse.go (hunk 1 above)
Comment (cameroncaci) [reply pkg/parser/tac/parse.go]: Yes it should, thank you for spotting that

File: pkg/parser/tac/parse.go (hunk 2 above)
Comment (cameroncaci) [reply pkg/parser/tac/parse.go]: Not quite for this spot. If the existing code has a blank transaction value, having the nested if statement prevents the delimiter ". Additional description found: " from being at the start of the new code's transaction

Comment (deandreJones) [reply pkg/parser/tac/parse.go]: ah.. good point, thnks for not just making the change based on suggestion

**Review by deandreJones — DISMISSED** [review]

**Review by j-Rosenthal — DISMISSED** [review]

**Review by danieljordan-caci — DISMISSED** [review]

**Review by TevinAdams — DISMISSED** [review]

**Comment by j-Rosenthal** [comment]
Conflicts

**Review by j-Rosenthal — APPROVED** [review]

**Review by deandreJones — APPROVED** [review]

**Pull request merged**